gui/src/installer/step/mod.rs

```diff
@@ -229,6 +638,71 @@ impl From<DefineBitcoind> for Box<dyn Step> {
     }
 }
 
+impl From<StartInternalBitcoindStep> for Box<dyn Step> {
+    fn from(s: StartInternalBitcoindStep) -> Box<dyn Step> {
+        Box::new(s)
+    }
+}
+
+impl StartInternalBitcoindStep {
+    pub fn new(liana_datadir: &PathBuf) -> Self {
+        Self {
+            conf_path: internal_bitcoind_config_path(liana_datadir),
+            network: Network::Bitcoin,
+            started: None,
+            exe_path: form::Value::default(),
+        }
+    }
+}
+
+impl Step for StartInternalBitcoindStep {
+    fn load_context(&mut self, ctx: &Context) {
+        if self.exe_path.value.is_empty() {
+            self.exe_path.value = bitcoind_exe_path();
+        }
+        self.network = ctx.bitcoin_config.network;
+    }
+    fn update(&mut self, message: Message) -> Command<Message> {
+        if let Message::StartInternalBitcoind(msg) = message {
+            match msg {
+                message::StartInternalBitcoindMsg::ExePathEdited(exe_path) => {
+                    self.exe_path.value = exe_path;
+                    self.exe_path.valid = PathBuf::from(self.exe_path.value.clone()).exists();
+                }
+                message::StartInternalBitcoindMsg::Start => {
+                    let args = vec![
+                        format!("-chain={}", self.network.to_core_arg()),
+                        format!("-conf={}", self.conf_path.to_string_lossy()),
+                    ];
+                    // TODO: How to run in Windows as daemon-like process?
```

Comment:
I don't think you can. You'd have to run it in its own thread. Pretty much how the GUI is running the Liana daemon right now if i'm not mistaken @edouardparis?

Reply:
GUI is running the daemon in a std::thread by calling directly the rust function.

Reply:
I've changed this slightly in https://github.com/wizardsardine/liana/commit/87509e9b4beeecbc49b618c72b3940c284ff345c so that it creates a child process and doesn't wait for the output. I could make some Unix-specific behaviour, but for now it's the same in both cases.

In order to check it's running fine, perhaps it would be worth proceeding to the `DefineBitcoind` step to check the connection, but with the cookie path and address forms hidden and only the "Check connection" button visible. Or I could try to reuse the `ping()` function in this same step.

Reply:
> I could make some Unix-specific behaviour, but for now it's the same in both cases.

In my opinion we should have bitcoind keep running as a daemon on Unix systems. But let's keep this for a follow-up. We can release Liana v2 with the bitcoind running in a thread that is stopped when Liana is stopped, get feedback and discuss in an issue letting it run as a daemon on Unix, and implement it for v3.

> [..] Checking the connection to bitcoind

We already have implemented those checks for the daemon. It could make sense to reuse the `BitcoinD` structure from there which performs sanity checks when instantiated?
https://github.com/wizardsardine/liana/blob/b205a25ed9eddcf4fea19efc790c27fb816dca02/src/bitcoin/d/mod.rs#L225-L230

I think this would require making the `bitcoin/d` module from the Liana daemon public, which is fine imo.

Reply:
I've updated to use `BitcoinD` to check the connection in https://github.com/wizardsardine/liana/commit/7ced7475cd93ec5dc0ca4c2dcd880f33812fa6f2.

This is using my Liana branch https://github.com/jp1ac4/liana/tree/make-bitcoind-public, which is branched from the commit used by the GUI (https://github.com/wizardsardine/liana/blob/master/gui/Cargo.lock#L1922).